lib/private/Share20/DefaultShareProvider.php

```diff
@@ -670,8 +670,21 @@ public function getSharesInFolder($userId, Folder $node, $reshares) {
 			);
 		}
 
+		// todo? maybe get these from the oc_mounts table
+		$childMountNodes = array_filter($node->getDirectoryListing(), function(Node $node) {
+			return $node->getInternalPath() === '';
+		});
+		$childMountRootIds = array_map(function(Node $node) {
+			return $node->getId();
+		}, $childMountNodes);
```

Comment:
```suggestion
		$childMountNodes = array_filter(
		    $node->getDirectoryListing(), 
		    fn(Node $node) => ($node->getInternalPath() === '')
		);
		$childMountRootIds = array_map(
		    fn(Node $node) => $node->getId(),
		    $childMountNodes
		);
```
Easier to read, harder to backport, just a suggestion.

Reply:
I prefer this with the "easier to backport" for now